test/livebook/zta/cloudflare_test.exs

```diff
@@ -0,0 +1,135 @@
+      "name" => "Tuka Peralta",
+      "email" => "tuka@peralta.com"
+    }
+
+    Bypass.expect_once(setup.bypass, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(200, Jason.encode!(%{keys: [setup.key]}))
+    end)
+
+    Bypass.expect_once(setup.user_identity, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(200, Jason.encode!(expected_user))
+    end)
+
+    {:ok, pid} = GenServer.start_link(Cloudflare, setup.options)
+
+    user = GenServer.call(pid, {:authenticate, [setup.token], fields: setup.fields})
+
+    assert %{id: "1234567890", email: "tuka@peralta.com", name: "Tuka Peralta"} = user
+  end
+
+  test "returns nil when the user_identity fails", setup do
+    Bypass.expect_once(setup.bypass, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(200, Jason.encode!(%{keys: [setup.key]}))
+    end)
+
+    Bypass.expect_once(setup.user_identity, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(403, "")
+    end)
+
+    {:ok, pid} = GenServer.start_link(Cloudflare, setup.options)
+
+    assert GenServer.call(pid, {:authenticate, [setup.token], fields: setup.fields}) == nil
+  end
+
+  test "returns nil when the identity is invalid", setup do
+    identity = %{
+      iss: "invalid",
+      key: "livebookweb",
+      user_identity: "http://localhost:#{setup.user_identity.port}"
+    }
+
+    options = Keyword.put(setup.options, :identity, identity)
+
+    Bypass.expect_once(setup.bypass, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(200, Jason.encode!(%{keys: [setup.key]}))
+    end)
+
+    {:ok, pid} = GenServer.start_link(Cloudflare, options)
+
+    assert GenServer.call(pid, {:authenticate, [setup.token], fields: setup.fields}) == nil
+  end
+
+  test "returns nil when the key is invalid", setup do
+    Bypass.expect_once(setup.bypass, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(200, Jason.encode!(%{keys: ["invalid_key"]}))
+    end)
+
+    {:ok, pid} = GenServer.start_link(Cloudflare, setup.options)
+
+    assert GenServer.call(pid, {:authenticate, [setup.token], fields: setup.fields}) == nil
+  end
+
+  test "returns nil when the token is invalid", setup do
+    Bypass.expect_once(setup.bypass, fn conn ->
+      conn
+      |> put_resp_content_type("application/json")
+      |> send_resp(200, Jason.encode!(%{keys: [setup.key]}))
+    end)
+
+    {:ok, pid} = GenServer.start_link(Cloudflare, setup.options)
+
+    assert GenServer.call(pid, {:authenticate, ["invalid_token"], fields: setup.fields}) == nil
+    assert GenServer.call(pid, {:authenticate, "invalid_token", fields: setup.fields}) == nil
+    assert GenServer.call(pid, {:authenticate, [], fields: setup.fields}) == nil
+    assert GenServer.call(pid, {:authenticate, nil, fields: setup.fields}) == nil
```

Comment:
I would prefer to not call the GenServer API directly. Can we instead write these tests by using `Plug.Test` to create a fake conn and put the header in the conn and then call the "official" authenticate API?